gdal/port/cpl_odbc.cpp

```diff
@@ -412,6 +412,63 @@ int CPLODBCSession::Failed( int nRetCode, HSTMT hStmt )
     return TRUE;
 }
 
+/************************************************************************/
+/*                          ConnectToMsAccess()                          */
+/************************************************************************/
+
+
+bool CPLODBCSession::ConnectToMsAccess(const char *pszName, const char *pszDSNStringTemplate)
+{
+    char *pszDSN = nullptr;
+
+    const bool usingAutomaticDSNStringTemplate = pszDSNStringTemplate == nullptr;
+
+    if( usingAutomaticDSNStringTemplate )
+    {
+#ifdef WIN32
+        pszDSNStringTemplate = "DRIVER=Microsoft Access Driver (*.mdb, *.accdb);DBQ=%s";
+#else
+        pszDSNStringTemplate = "DRIVER=Microsoft Access Driver (*.mdb, *.accdb);DBQ=\"%s\"";
+#endif
+    }
+    pszDSN = (char *) CPLMalloc(strlen(pszName)+strlen(pszDSNStringTemplate)+100);
```

Comment:
code in port/ no longer allows C style casts. So either use static_cast<char*> (here and at line 451). Or just go for std::string